### PR TITLE
ItemSample: Allow toggle of visibility via mouse click in LegendItem

### PR DESCRIPTION
**Pull request by dgoeries**
This is a very simple implementation of a strong feature. We allow to **toggle the visibility** of the data items via a mouse click on the **ItemSample** in the **LegendItem**.

In the beginning I tried to run for a nice svg icon, but had to realize that icons are not really welcome in this package (package data).

Hence, I went for a fairly neutral but clear implementation with visibility.

### CODE REVIEW

**Comment by dgoeries** [comment]
legend_toggle

**Comment by j9ac9k** [comment]
@dgoeries I will gladly assist/contribute to this PR w/ the packaging element if you want to try and get some nicer icons in there.

**Comment by dgoeries** [comment]
I added a possible implementation. My proposal would be to change the pixmaps in a next PR

File: pyqtgraph/icons/__init__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+    @property
+    def qicon(self):
+        if self._icon is None:
+            self._icon = QtGui.QIcon(op.join(op.dirname(__file__), self._path))
+
+        return self._icon
+
+
+def get_graph_icon(name):
```
Comment (j9ac9k) [inline pyqtgraph/icons/__init__.py:27]: probably should use `getGraphIcon` to match camelCase convention elsewhere

File: setup.py [hunk 2]
```diff
@@ -139,7 +139,9 @@ def run(self):
               'style': helpers.StyleCommand},
     packages=allPackages,
     package_dir={'pyqtgraph.examples': 'examples'},  ## install examples along with the rest of the source
-    package_data={'pyqtgraph.examples': ['optics/*.gz', 'relativity/presets/*.cfg']},
+    package_data={'pyqtgraph.examples': ['optics/*.gz', 'relativity/presets/*.cfg'],
+                  "pyqtgraph.icons": ["*.*"],
```
Comment (j9ac9k) [inline setup.py:143]: this should be more restrictive than `*.*`, but likely be `*.svg`

File: pyqtgraph/icons/__init__.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    """Return a `PyQtGraph icon from the registry by `name`"""
+    icon = _ICON_REGISTRY[name]
+    if isinstance(icon, GraphIcon):
+        icon = icon.qicon
+        _ICON_REGISTRY[name] = icon
+
+    return icon
+
+# Note: List all graph icons here ...
+invisibleEye = GraphIcon("invisibleEye.svg")
```
Comment (j9ac9k) [inline pyqtgraph/icons/__init__.py:37]: I don't think this variable is used here; should we just have a commented out list?

Comment (dgoeries) [reply pyqtgraph/icons/__init__.py:37]: oki

File: pyqtgraph/icons/__init__.py (hunk 1 above)
Comment (dgoeries) [reply pyqtgraph/icons/__init__.py]: done

**Comment by j9ac9k** [comment]
@dgoeries this LGTM, merging, thanks for the PR, sorry it took me so long to get to this.

**Pull request merged**